modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/helper/ValidationHelper.java

```diff
@@ -31,6 +32,19 @@
 @Component(service = ValidationHelper.class)
 public class ValidationHelper {
 
+	public boolean isValidAPIApplicationCompanyId(long objectEntryId) {
+		ObjectEntry objectEntryAPIApplication =
+			_objectEntryLocalService.fetchObjectEntry(objectEntryId);
+
+		if (objectEntryAPIApplication.getCompanyId() !=
+				CompanyThreadLocal.getCompanyId()) {
```

Comment:
Maybe instead of using the `CompanyThreadLocal.getCompanyId()` we can use the `companyId` that comes from the `schemaObjectEntry`, what do you think?

Reply:
Done ! https://github.com/liferay-headless/liferay-portal/pull/1896/commits/48c52c989d80e5233ac9aabbc020c312a4cf0e37